keylime/tpm/ec_crypto_helper.py

```diff
@@ -0,0 +1,101 @@
+# pylint: disable=protected-access
+
+from typing import Optional
+
+from cryptography.hazmat import backends
+from cryptography.hazmat.backends.openssl.backend import Backend as OpenSSLBackend
+from cryptography.hazmat.primitives.asymmetric.ec import EllipticCurvePrivateKey, EllipticCurvePublicKey
+
+
+class EcCryptoHelper:
+    instance: Optional["EcCryptoHelper"] = None
+
+    @staticmethod
+    def get_instance() -> "EcCryptoHelper":
+        """Create and return a singleton EcCryptoHelper"""
+        if not EcCryptoHelper.instance:
+            EcCryptoHelper.instance = EcCryptoHelper()
+        return EcCryptoHelper.instance
```

Comment:
I'm wondering if we actually need a class and singleton pattern here. The backend could be just declared in the module and the rest are then normal functions. 

@aplanas what is the pythonic way to do this?

Reply:
There are other ways of doing this of course but the singleton is there to check the validity of the backend class only once...

Reply:
Not sure to understand why a single instance is needed. Is because the __init__ checks are too costly?